File: 백준/Silver/10828. 스택/스택.kt
package org.example

import java.util.StringTokenizer

fun main(args: Array<String>) {
    val br = System.`in`.bufferedReader()
    val sb = StringBuilder()
    val n = br.readLine().toInt()
    val stack = mutableListOf<Int>()
    repeat(n) {
        val st = StringTokenizer(br.readLine())
        when (st.nextToken()) {
            "push" -> {
                stack.add(st.nextToken().toInt())
            }

            "pop" -> {
                if (stack.isNotEmpty()) {
                    sb.append(stack.removeLast())
                } else {
                    sb.append(-1)
                }
                sb.append("\n")
            }

            "size" -> {
                sb.append(stack.size).append("\n")
            }

            "empty" -> {
                if (stack.isEmpty()) {
                    sb.append(1)
                } else {
                    sb.append(0)
                }
                sb.append("\n")
            }

            "top" -> {
                if (stack.isNotEmpty()) {
                    sb.append(stack.last())
                } else {
                    sb.append(-1)
                }
                sb.append("\n")
            }
        }
    }
    print(sb)
    br.close()
}

File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline

n = int(input())

stack = []
for i in range(n):
    commands = input().split()
    command = commands[0]
    num = -1
    if len(commands) > 1:
        num = commands[1]

    if command == "push":
        stack.append(num)
    elif command == "pop":
        if len(stack)==0:
            print(-1)
        else:
            print(stack.pop())
    elif command =="size":
        print(len(stack))
    elif command == "empty":
        if len(stack)==0:
            print(1)
        else:
            print(0)
    elif command =="top":
        if len(stack)==0:
            print(-1)
        else:
            print(stack[-1])
